src/socket.ts

```diff
@@ -168,4 +203,68 @@ export default class SocketServer {
       throw new SocketError(SOCKET_ERROR_TYPE.SOMETHING_WENT_WRONG);
     }
   }
+
+  async verifySocketConnect(token: string) {
+    try {
+      if (!token) {
+        throw new SocketError(UNAUTHORIZED_MESSAGE);
+      }
+
+      return await userService.findUserByToken(token);
+    } catch (error) {
+      throw new SocketError(SOCKET_ERROR_TYPE.USER_NOT_EXIST);
+    }
+  }
+
+  setSocketAuth({
+    user,
+    socket,
+  }: {
+    user: HydratedDocument<IUser>;
+    socket: socket.Socket;
+  }) {
+    socket.data.userId = user._id.toString();
+  }
+
+  async handleSocketConnect(userId: string) {
+    try {
+      console.log('connected user', userId);
+
+      let connectedSocket: string[] | null = await getRedisValue(
+        SOCKET_CONNECT_USERS,
+      );
+      console.log(connectedSocket);
+
+      if (!connectedSocket) {
+        setRedisValue(SOCKET_CONNECT_USERS, JSON.stringify([]));
+        connectedSocket = [];
+      }
+
+      if (!connectedSocket.includes(userId)) {
+        connectedSocket.push(userId);
+        setRedisValue(SOCKET_CONNECT_USERS, JSON.stringify(connectedSocket));
+      }
+    } catch (error) {
+      throw new SocketError(SOCKET_ERROR_TYPE.USER_NOT_EXIST);
+    }
+  }
+
+  async handleSocketDisconnect(userId: string) {
+    try {
+      console.log('disconnected user', userId);
+
+      const connectedSocket: string[] = await getRedisValue(
+        SOCKET_CONNECT_USERS,
+      );
+
+      const index = connectedSocket.findIndex((id) => userId === id);
```

Comment:
We can use filter for this one. it will make sure every duplicate ids are removed